src/execution/sql/table_vector_iterator.cpp

```diff
@@ -23,41 +31,99 @@ bool TableVectorIterator::Init() {
   // Find the table
   table_ = exec_ctx_->GetAccessor()->GetTable(table_oid_);
   TERRIER_ASSERT(table_ != nullptr, "Table must exist!!");
-
+  if (col_oids_.empty()) {
+    table_->GetAllColOid(&col_oids_);
+  }
   // Initialize the projected column
-  TERRIER_ASSERT(!col_oids_.empty(), "There must be at least one col oid!");
+  // TERRIER_ASSERT(!col_oids_.empty(), "There must be at least one col oid!");
   auto pc_init = table_->InitializerForProjectedColumns(col_oids_, common::Constants::K_DEFAULT_VECTOR_SIZE);
   buffer_ = exec_ctx_->GetMemoryPool()->AllocateAligned(pc_init.ProjectedColumnsSize(), alignof(uint64_t), false);
   projected_columns_ = pc_init.Initialize(buffer_);
   initialized_ = true;
 
-  // Begin iterating
-  iter_ = std::make_unique<storage::DataTable::SlotIterator>(table_->begin());
+  // Create the start and end iterator
+  iter_ = std::make_unique<storage::DataTable::SlotIterator>(table_->beginAt(start_block_idx_));
+  iter_end_ = std::make_unique<storage::DataTable::SlotIterator>(table_->endAt(end_block_idx_));
   return true;
 }
 
 bool TableVectorIterator::Advance() {
```

Comment:
This function now seems more like an `AdvanceToEnd` function

Reply:
> This function now seems more like an `AdvanceToEnd` function
The advance() function is still possible to be executed multiple times if the buffer is full during the previous round of scan. So it might not reach the end in this case.